spanner/client.go

```diff
@@ -280,6 +280,9 @@ func (c *Client) Single() *ReadOnlyTransaction {
 		t.sh = sh
 		return nil
 	}
+	if c.sc != nil && commonTags.clientID == "" {
```

Comment:
Sorry, I think I was a little unclear on what I meant with where the `commonTags` should go:

1. `commonTags` will never change for a given `sessionClient` or `Client`. So one of the `sessionClient` or `Client` structs should therefore have a field `commonTags` (or just the values that are needed for common tags). `commonTags` should not be a global variable. The current implementation means that `commonTags` will be filled with the values of the first client that executes a transaction. If the application opens a second client for a different database, `commonTags` will continue to contain the values for the first client.
2. The initialization of `commonTags` should not happen when a transaction is created. It should be done when the `Client` / `sessionClient` is created, as it will never change after that. So in this case somewhere around line 194 in this file. That means that these checks and `setCommonTags(..)` calls can be removed from all the transaction methods.